pkg/asset/installconfig/gcp/validation.go

```diff
@@ -172,35 +170,80 @@ func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.
 	return allErrs
 }
 
-// ValidatePreExitingPublicDNS ensure no pre-existing DNS record exists in the public
-// DNS zone for cluster's Kubernetes API.
-func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
+// ValidatePreExistingPrivateDNS ensure that the PrivateZone exists in the cluster domain
+func ValidatePreExistingPrivateDNS(client API, ic *types.InstallConfig) error {
+
+	if ic.Platform.GCP.PrivateDNSZone != nil && ic.Platform.GCP.PrivateDNSZone.ID != "" {
+		project := ic.Platform.GCP.PrivateDNSZone.ProjectID
```

Comment:
nit: the `ValidatePreExistingPublicDNS` function below defaults to the `ic.Platform.GCP.ProjectID`. You might want to choose one default project for both functions for consistency. My recommendation would be to use the `ic.Platform.GCP.ProjectID` as that would be congruent with the actual default behavior.
```
project := ic.Platform.GCP.PrivateDNSZone.ProjectID
if if.Platform.GCP.PublicDNSZone != nil && ic.Platform.GCP.PublicDNSZone.ProjectID != "" {
    project = ic.Platform.GCP.PublicDNSZone.ProjectID
}
```